lib/ts/recipe/thirdparty/recipeImplementation.ts

```diff
@@ -0,0 +1,262 @@
+/* Copyright (c) 2022, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import STGeneralError from "../../error";
+import Querier from "../../querier";
+import { NormalisedAppInfo } from "../../types";
+import { appendQueryParamsToURL, getQueryParams, getSessionStorage, setSessionStorage } from "../../utils";
+import { UserType } from "../authRecipeWithEmailVerification/types";
+import { NormalisedInputType, RecipeInterface, StateObject } from "./types";
+import { RecipeFunctionOptions } from "../recipeModule/types";
+import { generateThirdPartyProviderState } from "./utils";
+
+export default function getRecipeImplementation(recipeId: string, appInfo: NormalisedAppInfo): RecipeInterface {
+    const querier = new Querier(recipeId, appInfo);
+    return {
+        getOAuthState: function (): {
+            status: "OK";
```

Comment:
You don't need to have status: OK in get and set  result

Reply:
Refactored types to only have status if an API call is made (Added a TODO to https://github.com/supertokens/for-zenhub/issues/28 for other recipes)